### PR TITLE
Issue/7166 remove legacy signup

### PR DESCRIPTION
**Pull request by theck13**
Remove legacy signup code that has been superseded by the [new signup flow](https://github.com/wordpress-mobile/WordPress-Android/issues/6838) as described in https://github.com/wordpress-mobile/WordPress-Android/issues/7166.

### CODE REVIEW

File: WordPress/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -94,21 +94,6 @@
...
-            android:name=".ui.accounts.SignInActivity"
-            android:configChanges="orientation|keyboardHidden|screenSize"
-            android:theme="@style/SignInTheme"
-            android:windowSoftInputMode="adjustResize">
-            <intent-filter>
-                <action android:name="android.intent.action.VIEW" />
-                <category android:name="android.intent.category.DEFAULT" />
-                <category android:name="android.intent.category.BROWSABLE" />
-                <data
-                    android:host="signin"
```
Comment (hypest) [inline WordPress/src/main/AndroidManifest.xml]: As @maxme has suggested over Slack, let's check with our #helpshift friends whether they need/use this shortcut to have people logged in. WDYT?

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java [hunk 2]
```diff
@@ -421,10 +420,6 @@ public static void setLoginWizardStyleActive(boolean loginWizardActive) {
         }
     }
 
-    public static boolean isLoginWizardStyleActivated() {
-        return BuildConfig.LOGIN_WIZARD_STYLE_ACTIVE || getBoolean(UndeletablePrefKey.LOGIN_WIZARD_STYLE_ACTIVE, false);
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java]: Can we remove `UndeletablePrefKey.LOGIN_WIZARD_STYLE_ACTIVE` as well perhaps?

File: WordPress/src/main/java/org/wordpress/android/ui/AddQuickPressShortcutActivity.java [hunk 3]
```diff
@@ -113,8 +112,7 @@ public void onItemClick(AdapterView<?> arg0, View row, int position, long id) {
 
         } else {
             // no account, load new account view
-            Intent i = new Intent(AddQuickPressShortcutActivity.this, SignInActivity.class);
-            startActivityForResult(i, ADD_ACCOUNT_REQUEST);
+            ActivityLauncher.showSignInForResult(AddQuickPressShortcutActivity.this);
```
Comment (hypest) [inline WordPress/src/main/java/org/wordpress/android/ui/AddQuickPressShortcutActivity.java]: `ActivityLauncher.showSignInForResult` uses `RequestCodes.ADD_ACCOUNT` as a request code but `AddQuickPressShortcutActivity.java` expects ADD_ACCOUNT_REQUEST in its `onActivityResult`. Will this still work?

File: WordPress/src/main/AndroidManifest.xml (hunk 1 above)
Comment (theck13) [reply WordPress/src/main/AndroidManifest.xml]: I checked with #helpshift and it doesn't seem to be used so we can remove it.

File: WordPress/src/main/java/org/wordpress/android/ui/AddQuickPressShortcutActivity.java (hunk 3 above)
Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/AddQuickPressShortcutActivity.java]: I don't think so.  Good catch!  I updated the request code in 46c88c1032a3eb640dddaa617cb44f6c27a09f86.

File: WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java (hunk 2 above)
Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/prefs/AppPrefs.java]: Yes, we can.  I removed the preference key in 5e8d7b3d5332b6525bab30146b831a7e13a36b3e.

**Comment by hypest** [comment]
LGTM!

**Pull request merged**